xnet/net_windows.go

```diff
@@ -0,0 +1,33 @@
+// +build windows
+
+package xnet
+
+import (
+	"net"
+	"time"
+
+	"github.com/Microsoft/go-winio"
+)
+
+// Listen Wrapper for net.Listen and Windows named pipe
+func Listen(proto string, addr string) (net.Listener, error) {
+	// Ok this explanation has to go somewhere, so it's going here
+	// as of the commit that added this comment (use `git blame`) there is an
+	// issue with the way we do listen. basically, `unix` is hard-coded as the
+	// protocol for the control api, and `tcp` as the remote api. on windows,
+	// we need to use named pipes `npipe` as the control api instead.
+	// so, here, we just use a hack. if the protocol is says unix but we're on
+	// windows, just use a named pipe anyway. problem changed.
+	if proto == "unix" {
```

Comment:
Personally I don't think this hack belongs here.

Just make this a `ListenLocal` function that isn't passed a protocol. The windows version should use `winio.ListenPipe` and the unix version should use `Listen("unix", addr)`.

For now the manager can special-case "unix" as needing a call to `ListenLocal`, but I think we should immediately refactor the manager to remove the `ProtoAddr` map so the hack isn't necessary.